Code/Framework/AzCore/AzCore/EBus/Policies.h

```diff
@@ -251,29 +250,29 @@ namespace AZ
         void Execute()
         {
             AZ_Warning("System", m_isActive, "You are calling execute queued functions on a bus which has not activated its function queuing! Call YourBus::AllowFunctionQueuing(true)!");
+            // The while true loop is used to allow for functions to be added to the message queue
+            // during the middle of executing the previous queue
             while (true)
             {
```

Comment:
Maybe this outer `while(true)` loop should be removed.  It seems like a nice feature to allow for additional functions to be added, but it means there's a side effect that `Execute()` completion time may get extended indefinitely.

I think a single lock-and-swap of the queue at the point when `Execute()` is called and just execute those functions is a clearer behavior, and new functions can still fill up the queue, but will be for "next turn".

Reply:
Agreed.

I have updated the logic to perform a single lock-and-swap